File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.java
class Solution {
    public int longestSubarray(int[] nums) {
        int mx = -1, count = 0, ans = 0;
        for(int it : nums) mx = Math.max(mx, it);

        for(int it : nums) {
            if(mx == it) count++;
            else count = 0;
            ans = Math.max(ans, count);
        }

        return ans;
    }
}

File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.py
class Solution(object):
    def longestSubarray(self, nums):
        mx, count, ans = 0, 0, 0
        for n in nums:
            mx = max(n, mx)
        
        for n in nums:
            if mx == n:
                count += 1
            else:
                count = 0
            ans = max(ans, count)
        
        return ans
        